workflow/algorithm.go

```diff
@@ -0,0 +1,124 @@
+package workflow
+
+import "fmt"
+
+// ChildrenIDs returns the list of node IDs with a dependency to the current node
+func (w Workflow) ChildrenIDs(nodeKey string) []string {
+	nodeKeys := make([]string, 0)
+	for _, edge := range w.Edges {
+		if edge.Src == nodeKey {
+			nodeKeys = append(nodeKeys, edge.Dst)
+		}
+	}
+	return nodeKeys
+}
+
+// ParentIDs returns the list of node IDs with the current node as child
+func (w Workflow) ParentIDs(nodeKey string) []string {
+	nodeKeys := make([]string, 0)
+	for _, edge := range w.Edges {
+		if edge.Dst == nodeKey {
+			nodeKeys = append(nodeKeys, edge.Src)
+		}
+	}
+	return nodeKeys
+}
+
+// FindNode returns the node matching the key in parameter or an error if not found
+func (w Workflow) FindNode(key string) (Node, error) {
+	for _, node := range w.Nodes {
+		if node.Key == key {
+			return node, nil
+		}
+	}
+	return Node{}, fmt.Errorf("node %q not found", key)
+}
+
+// A null graph is a graph that contains no nodes
+func (w *Workflow) hasNodes() bool {
+	return len(w.Nodes) == 0
```

Comment:
The bool return is opposite of the name of the function...
on empty graph: hasNodes true?. it should be false.
```suggestion
	return len(w.Nodes) > 0
```